### ISSUE TITLE
LightGBM predictor native-artifact stall on nv0 exceeds 60 s for small temporal dataset

### ISSUE THREAD

**Issue by nm-z** [nm-z/recipe-dev#503]
Public `recipe.train().optimizer(adamw).lr(0.0001).seed(114367219561025).epochs(1).log(all).stop(0.8).fp(32).run(&model, &data)` on backend `nv0` remains active past 60 s. The data is `recipe.data("data/temporal/series_sqlite.sqlite").target("target").norm(z_score)`; the earliest model block is `.lgbm().prelu().qi(3).k.s` followed by `.conv(8, 1).norm(batch).iq(4).nl`. The earliest failing operation is `recipe.train().run()` crossing the slow-runtime threshold; the inspected path is `.lgbm()` -> `fit_lightgbm` -> `boosted_predictor` -> `NativeModelIr::from_graph` -> `compile_native_artifact` (NVIDIA `ptx`/`ptxas` branch), where the predictor primitive compilation for the quantized LightGBM trees stalls. The arithmetic workload (default `boost_iterations=8`, small `lightgbm_leaves`, a few hundred temporal rows/features, one `conv(8,1)` contraction) and memory traffic justify a sub-second epoch on nv0; the 60 s active state is materially slower. No existing open issue covers the `.lgbm()` predictor path on nv0 (#481 is CPU only, #497 is XGBoost, #502 is attention), so this is a distinct performance defect.

## Machine provenance

- Discoverer: Recipe exhaustive Cartesian traversal
- Failure fingerprint: `39b6ead4072e7139`
- Classifier provider: `openrouter`
- Classifier model: `thinkingmachines/inkling:free`
- Classifier effort: `default`

### Classifier decision

```json
{"verdict":"new","issue":0,"title":"LightGBM predictor native-artifact stall on nv0 exceeds 60 s for small temporal dataset","body":"Public `recipe.train().optimizer(adamw).lr(0.0001).seed(114367219561025).epochs(1).log(all).stop(0.8).fp(32).run(&model, &data)` on backend `nv0` remains active past 60 s. The data is `recipe.data(\"data/temporal/series_sqlite.sqlite\").target(\"target\").norm(z_score)`; the earliest model block is `.lgbm().prelu().qi(3).k.s` followed by `.conv(8, 1).norm(batch).iq(4).nl`. The earliest failing operation is `recipe.train().run()` crossing the slow-runtime threshold; the inspected path is `.lgbm()` -> `fit_lightgbm` -> `boosted_predictor` -> `NativeModelIr::from_graph` -> `compile_native_artifact` (NVIDIA `ptx`/`ptxas` branch), where the predictor primitive compilation for the quantized LightGBM trees stalls. The arithmetic workload (default `boost_iterations=8`, small `lightgbm_leaves`, a few hundred temporal rows/features, one `conv(8,1)` contraction) and memory traffic justify a sub-second epoch on nv0; the 60 s active state is materially slower. No existing open issue covers the `.lgbm()` predictor path on nv0 (#481 is CPU only, #497 is XGBoost, #502 is attention), so this is a distinct performance defect.","rationale":"Read-only inspection of `recipe.rs` confirms `.lgbm()` lowers through `fit_lightgbm` (line 7437) to `boosted_predictor` (line 7344), emitting a `PredictorOpcode` program for the fitted trees. On `nv0`, `compile_model` (line 2868) emits the native artifact through `compile_native_artifact` (line 2828) using the NVIDIA `ptx`/`ptxas` compiler. The measured 60-second active run on a small temporal SQLite dataset with minimal arithmetic work is unjustified; the dominant gap is the NVIDIA native-artifact compilation/execution stall for the `.lgbm()` predictor, not the `.conv()` block or loader. This earliest cause (LightGBM predictor on nv0) is not represented by any existing open issue, so the packet is `new`."}
```

## Deterministic failure packet

kind=performance
id=39b6ead4072e7139
base=commit=9cc65a63b4677c53372379219cb280da9085ac1b tracked_tree=clean
cursor=seed:0 cursor:14395 next:14396 composition:114367219561025
expected=one training epoch uses throughput proportionate to its operations, data volume, memory traffic, and available hardware
observed=the public Recipe run remained active after 60 seconds
measurement=elapsed_seconds:60 backend:nv0
replay=the configured slow-runtime threshold was crossed stable:true
command=cargo run --bin recipe -- /home/nate/Desktop/recipe-machine-workload/target/debug/recipe-trials/recipe-trial-185-14395/reproduction.rs
reproduction:
```rust
use recipe::*;

fn main() {
	let bundle = format!("/tmp/recipe-composition-repro-{}.ogdl", std::process::id());
	let data = recipe.data("data/temporal/series_sqlite.sqlite")
		.target("target")
		.norm(z_score);
	let model = recipe.model()
		.lgbm().prelu().qi(3).k.s
		.conv(8, 1).norm(batch).iq(4).nl
		.loss(focal);
	let report = recipe.train()
		.optimizer(adamw)
		.lr(0.0001)
		.seed(114367219561025)
		.epochs(1)
		.log(all)
		.stop(0.8)
		.fp(32)
		.save(&bundle)
		.run(&model, &data);
	assert!(report.final_loss().is_finite());
}
```
backend=nv0

**Comment by nm-z** [comment]
The public `recipe.train().run()` remains active beyond 60 seconds while `compile_model` emits and compiles the `Primitive::Predictor` produced by `lower_estimator` through `fit_lightgbm` and `boosted_predictor` on the NVIDIA `ptx`/`ptxas` path. This is additional evidence for the same LightGBM native-artifact stall.

## Machine provenance

- Discoverer: Recipe exhaustive Cartesian traversal
- Failure fingerprint: `d01b91963cd7e1ed`
- Classifier provider: `copilot`
- Classifier model: `gpt-5.6-luna`
- Classifier effort: `auto`

### Classifier decision

```json
{
  "verdict": "comment",
  "issue": 503,
  "title": "LightGBM predictor native-artifact stall on nv0 exceeds 60 s for small temporal dataset",
  "body": "The public `recipe.train().run()` remains active beyond 60 seconds while `compile_model` emits and compiles the `Primitive::Predictor` produced by `lower_estimator` through `fit_lightgbm` and `boosted_predictor` on the NVIDIA `ptx`/`ptxas` path. This is additional evidence for the same LightGBM native-artifact stall.",
  "rationale": "Issue #503 represents the same earliest public failure and implementation path. The temporal SQLite workload loads about 12,000 rows and 1 feature. The source config permits 8 boosting iterations, 8 leaves per tree, up to 7 splits per tree with maximum depth 7, and 16 histogram candidates per feature per active leaf. This is roughly 10 million host-side histogram operations, about 1.4 million predictor comparisons for 12,000 rows, and only tens of megabytes of traffic, followed by a small pool operation. That workload should complete in well under a few seconds on the selected nv0 GPU path, whose exact device capability is not exposed by the packet. The measured 60 seconds is therefore materially unjustified, with the dominant gap being NVIDIA native artifact compilation, not LightGBM fitting, pooling, loss, or quantization configuration."
}
```

## Deterministic failure packet

kind=performance
id=d01b91963cd7e1ed
base=commit=9cc65a63b4677c53372379219cb280da9085ac1b tracked_tree=modified
cursor=seed:0 cursor:24053 next:24054 composition:32791185470287
expected=one training epoch uses throughput proportionate to its operations, data volume, memory traffic, and available hardware
observed=the public Recipe run remained active after 60 seconds
measurement=elapsed_seconds:60 backend:nv0
replay=the configured slow-runtime threshold was crossed stable:true
command=cargo run --bin recipe -- /home/nate/Desktop/recipe-machine-workload/target/debug/recipe-trials/recipe-trial-8759-24053/reproduction.rs
reproduction:
```rust
use recipe::*;

fn main() {
	let bundle = format!("/tmp/recipe-composition-repro-{}.ogdl", std::process::id());
	let data = recipe.data("data/temporal/series_sqlite.sqlite")
		.target("target")
		.broadcast();
	let model = recipe.model()
		.lgbm().sigmoid().qi(3).k.l
		.pool(2).exp().iq(2).m
		.loss(huber);
	let report = recipe.train()
		.optimizer(adamw)
		.lr(0.0001)
		.seed(32791185470287)
		.epochs(1)
		.log(all)
		.fp(32)
		.save(&bundle)
		.run(&model, &data);
	assert!(report.final_loss().is_finite());
}
```
backend=nv0
